src/registry/utils.ts

```diff
@@ -15,8 +17,25 @@ import {
 } from "../helpers/ethers.js";
 import { validateTables, type StatementType } from "../helpers/parser.js";
 
+/**
+ * UnnamedWaitableTransactionReceipt represents a TransactionReceipt with a wait method, but no name.
+ */
+export type UnnamedWaitableTransactionReceipt = TransactionReceipt &
+  Wait<TransactionReceipt>;
+
 /**
  * WaitableTransactionReceipt represents a named TransactionReceipt with a wait method.
+ * See the Validator spec in the docs for more details.
+ * @typedef {Object} WaitableTransactionReceipt
+ * @property {function} wait - Async function that will not return until the validator has processed tx.
+ * @property {string} name - The full table name.
+ * @property {string} prefix - The table name prefix.
+ * @property {number} chainId - The chainId of tx.
+ * @property {string} tableId - The tableId of tx.
+ * @property {string} transaction_hash - The transaction hash of tx.
+ * @property {number} block_number - The block number of tx.
+ * @property {Object} error - The first error encounntered when the Validator processed tx.
+ * @property {number} error_event_idx - The index of the event that cause the error when the Validator processed tx.
```

Comment:
Since this type is generated from the api spec, but because our api has [a bug](https://github.com/tablelandnetwork/go-tableland/issues/522), it's kind of hard to reason about what this actually looks like.
I added this here for to help with readability in the short term.  We might want to remove this comment in the future since the definition comes from a different repo and keeping this in sync with that could be prone to oversight.